src/js/components/ServiceForm.js

```diff
@@ -0,0 +1,225 @@
+import {Tooltip} from 'reactjs-components';
+import React from 'react';
+
+import FormUtil from '../utils/FormUtil';
+import GeminiUtil from '../utils/GeminiUtil';
+import SchemaFormUtil from '../utils/SchemaFormUtil';
+import SchemaUtil from '../utils/SchemaUtil';
+import TabForm from './TabForm';
+import Util from '../utils/Util';
+
+const METHODS_TO_BIND = [
+  'getAddNewRowButton',
+  'getRemoveVariableButton',
+  'getTriggerTabFormSubmit',
+  'handleFormChange',
+  'handleExternalSubmit',
+  'validateForm'
+];
+
+class ServiceForm extends React.Component {
+  constructor() {
+    super();
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+    this.triggerSubmit = function () {};
+    this.isValidated = true;
+  }
+
+  componentWillMount() {
+    this.multipleDefinition = this.getNewDefinition();
+    this.props.getTriggerSubmit(this.handleExternalSubmit);
+  }
+
+  componentWillUnmount() {
+    // Unschedule all validation if component unmounts.
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+  }
+
+  componentDidUpdate() {
+    // Timeout necessary due to modal content height updates on did mount
+    setTimeout(() => {
```

Comment:
Let's use `requstAnimationFrame` as we want to wait until stuff is render. and the modal height is updated.